larq_compute_engine/mlir/transforms/prepare_patterns.td

```diff
@@ -52,12 +54,15 @@ def : Pat<(TF_Conv2DOp: $output
             (TF_TransposeOp (TF_DivOp (ConstantOp $filter), (ConstantOp (GetScaleVector $filter))), (ConstantOp ConstantAttr<I32VectorElementsAttr<4>, "{3, 0, 1, 2}">)),
             (ConstantOp (GetScaleVector $filter)),
             (ConstantOp (GetConstantVector<"0.0f"> $filter)),
+            ConstantAttr<I32Attr, "-1">,
```

Comment:
Do you think it would be nicer to directly assign the shape here in tablegen using something like [`this`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/mlir/lite/transforms/optimize_patterns.td#L396) and [`this`](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/mlir/lite/transforms/optimize.cc#L193-L207)?

If not, feel free to ignore this comment :)

Reply:
@AdamHillier How do you feel about always setting `channel_in`? Another way would be to set the default to `None` wihch might be cleaner.

Reply:
Done in 09ea77a.